src/main/java/lambdaperson/FilterPersonsService.java

```diff
@@ -0,0 +1,18 @@
+package lambdaperson;
+
+import personList.Person;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class FilterPersonsService {
+
+    public List<String> filter(Set<Person> personList) {
+        return personList.stream().filter(person -> person.getAge() < 30 && person.getAge() > 20)
+                .peek((Person p) -> p.setName(p.getName().toUpperCase()))
```

Comment:
It seems I've missed that, sorry.

It's a bad practice to mutate a stream element in stream processing. Much better way is to create a Person with adjusted names.
But in the next step there's `map(Person::getName)`, so that means that there's no need to change `Person#name` at all: after `filter` one can make `map(Person::getName)`, then map the resulting `name` with `toUpperCase` and then sort then by lenght.